Doc/library/ast.rst

```diff
@@ -146,6 +146,95 @@ Node classes
     Snakes <https://greentreesnakes.readthedocs.io/en/latest/>`__ project and
     all its contributors.
 
+
+.. _ast-root-nodes:
+
+Root nodes
+^^^^^^^^^^
+
+.. class:: Module(body, type_ignores)
+
+   A Python module.
+
+   *body* is a :class:`list` of the module's :ref:`ast-statements`.
+   *type_ignores* is a :class:`list` of the module's type ignore comments,
+   see :func:`ast.parse` for more details.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('x = 1'), indent=4))
+        Module(
+            body=[
+                Assign(
+                    targets=[
+                        Name(id='x', ctx=Store())],
+                    value=Constant(value=1))],
+            type_ignores=[])
+
+
+.. class:: Expression(body)
+
+   A single Python expression.
+
+   *body* is a single :ref:`ast-expressions` node.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('123', mode='eval'), indent=4))
+        Expression(
+            body=Constant(value=123))
+
+
+.. class:: Interactive(body)
+
+   A single Python interactive statement, like in :ref:`tut-interac`.
+
+   *body* is a :class:`list` of :ref:`ast-statements` nodes.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('x = 1; y = 2', mode='single'), indent=4))
+        Interactive(
+            body=[
+                Assign(
+                    targets=[
+                        Name(id='x', ctx=Store())],
+                    value=Constant(value=1)),
+                Assign(
+                    targets=[
+                        Name(id='y', ctx=Store())],
+                    value=Constant(value=2))])
+
+
+.. class:: FunctionType(argtypes, returns)
+
+   Special type to represent old-style type comments for functions,
+   as Python versions prior to 3.5 didn't support :pep:`484` annotations.
+
+   Such type comments would look like this::
+
+       def sum_two_number(a, b):
+           # type: (int, int) -> int
+           return a + b
+
+   *argtypes* is a :class:`list` of :ref:`ast-expressions` nodes.
+   *returns* is a single :ref:`ast-expressions` node.
```

Comment:
```suggestion
   *argtypes* is a :class:`list` of :ref:`expression nodes <ast-expressions>`.

   *returns* is a single :ref:`expression node <ast-expressions>`.
```

* Make this grammatical
* Clarity that we are linking to expression node types rather than the `ast.Expression` node
* Put separate parameters on separate lines for scannability